test/functional/tests/cache_ops/test_cleaning_policy_operation.py

```diff
@@ -0,0 +1,284 @@
+
+    with TestRun.step("Check for running CAS cleaner"):
+        if TestRun.executor.run("pgrep cas_clean").exit_code != 0:
+            TestRun.fail("CAS cleaner process is not running!")
+
+    with TestRun.step(f"Add {cores_count} cores to the cache"):
+        core = []
+        for i in range(cores_count):
+            core.append(cache.add_core(core_dev.partitions[i]))
+
+    with TestRun.step("Run 'fio'"):
+        for i in range(cores_count):
+            fio_prepare(core[i]).run()
+        sync()
+        core_writes_before_wait_for_cleaning = (
+            cache.get_statistics().block_stats.core.writes
+        )
+
+    with TestRun.step(f"Wait {time_to_wait} seconds"):
+        time.sleep(time_to_wait)
+
+    with TestRun.step("Check write statistics for core device"):
+        core_writes_after_wait_for_cleaning = (
+            cache.get_statistics().block_stats.core.writes
+        )
+        check_cleaning_policy_operation(
+            cleaning_policy,
+            core_writes_before_wait_for_cleaning,
+            core_writes_after_wait_for_cleaning,
+        )
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+        Udev.enable()
+
+
+@pytest.mark.parametrize("cleaning_policy", CleaningPolicy)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_cleaning_policies_in_write_through(cleaning_policy):
+    """
+        title: Test for cleaning policy operation in Write-Through cache mode.
+        description: |
+          Check if ALRU, NOP and ACP cleaning policies preserve their
+          parameters when changed and if they flush dirty data properly
+          in Write-Through cache mode.
+        pass_criteria:
+          - Flush parameters preserve their values when changed.
+          - Dirty data is flushed or not according to the policy used.
+    """
+
+    with TestRun.step("Partition cache and core devices"):
+        cache_dev, core_dev = storage_prepare()
+        Udev.disable()
+
+    with TestRun.step(
+        f"Start cache in Write-Through mode with {cleaning_policy} cleaning policy"
+    ):
+        cache = casadm.start_cache(cache_dev.partitions[0], CacheMode.WT, force=True)
+        set_cleaning_policy_and_params(cache, cleaning_policy)
+
+    with TestRun.step("Check for running CAS cleaner"):
+        if TestRun.executor.run("pgrep cas_clean").exit_code != 0:
+            TestRun.fail("CAS cleaner process is not running!")
+
+    with TestRun.step(f"Add {cores_count} cores to the cache"):
+        core = []
+        for i in range(cores_count):
+            core.append(cache.add_core(core_dev.partitions[i]))
+
+    with TestRun.step("Change cache mode to Write-Back"):
+        cache.set_cache_mode(CacheMode.WB)
+
+    with TestRun.step("Run 'fio'"):
+        for i in range(cores_count):
+            fio_prepare(core[i]).run()
```

Comment:
We may want to run `fio` in parallel on all cores.

Reply:
Done